ui/app/templates/components/clients/dashboard.hbs

```diff
@@ -70,162 +50,148 @@
+              @dataTwoData="4"
             />
           {{/if}}
-        {{else}}
-          <EmptyState @title="No data received" @message="No data exists for that query period. Try searching again." />
         {{/if}}
+      {{else}}
+        <EmptyState
+          @title="No billing start date found"
+          @message="In order to get the most from this data, please enter your billing period start month. This will ensure that the resulting data is accurate."
+        />
       {{/if}}
-    {{/if}}
-  </div>
 
-  {{! Modal for startDate picker }}
-  <Modal
-    @title="Edit start month"
-    @onClose={{action (mut this.isEditStartMonthOpen) false}}
-    @isActive={{this.isEditStartMonthOpen}}
-    @showCloseButton={{true}}
-  >
-    <section class="modal-card-body">
-      <p class="has-bottom-margin-s">
-        This date is when client counting starts, or the point at which clients become unique. Without this starting point,
-        the data shown is not reliable.
-      </p>
-      <p class="has-bottom-margin-s"><strong>Billing contract start month</strong></p>
-      <div class="modal-radio-button">
-        <BasicDropdown @class="popup-menu" @horizontalPosition="auto-right" @verticalPosition="below" as |D|>
-          <D.Trigger
-            data-test-popup-menu-trigger="true"
-            class={{concat "toolbar-link" (if D.isOpen " is-active")}}
-            @htmlTag="button"
+      {{! Modal for startDate picker }}
+      <Modal
+        @title="Edit start month"
+        @onClose={{action (mut this.isEditStartMonthOpen) false}}
+        @isActive={{this.isEditStartMonthOpen}}
+        @showCloseButton={{true}}
+      >
+        <section class="modal-card-body">
+          <p class="has-bottom-margin-s">
+            This date is when client counting starts, or the point at which clients become unique. Without this starting
+            point, the data shown is not reliable.
+          </p>
+          <p class="has-bottom-margin-s"><strong>Billing contract start month</strong></p>
+          <div class="modal-radio-button">
+            <BasicDropdown @class="popup-menu" @horizontalPosition="auto-right" @verticalPosition="below" as |D|>
+              <D.Trigger
+                data-test-popup-menu-trigger="true"
+                class={{concat "toolbar-link" (if D.isOpen " is-active")}}
+                @htmlTag="button"
+              >
+                {{or this.startMonth "Month"}}
+                <Chevron @direction="down" @isButton={{true}} />
+              </D.Trigger>
+              <D.Content class="popup-menu-content is-wide">
+                <nav class="box menu scroll">
+                  <ul class="menu-list">
+                    {{#each this.months as |month|}}
+                      <button
+                        type="button"
+                        class="link"
+                        onclick={{queue (action "selectStartMonth" month) (action D.actions.close)}}
```

Comment:
I think it's recommended in Octane to use the `on` modifier to attach event handlers to HTML elements. `{{on "click" ...}}`

Reply:
I'll work with this. The queue part her is key and I was having problems using the on modifier. But maybe with fresh eyes I can get it.

Reply:
Got it - key was to put queue after the on "click" which was an easy see with fresh eyes. `{{on "click" (queue (fn this.selectStartMonth month) (action D.actions.close))}}`